Google-Summer-of-Code-2023.md

```diff
@@ -273,3 +273,504 @@ The following is a list of Oppia's 2023 GSoC project ideas. You are welcome to c
+* Should use Docker compose V2 (since compose V1 has been [deprecated](https://www.docker.com/blog/announcing-compose-v2-general-availability/)).
+
+**Size of this project:** large (~350 hours)
+
+**Potential Mentors:** 
+
+**Knowledge/Skills Recommended:**
+
+**Suggested Milestones:**
+
+* **Milestone 1:** The server launch should be successful and all Oppia pages should be functioning correctly.
+
+* **Milestone 2:** All tests should be executed locally and the migration of GitHub actions to utilize docker, including caching of docker images, should be completed.
+
+**Dependency on Release Schedule:** 
+
+**Proposal notes:**
+
+**Useful resources:**
+
+
+### 4.2. Developer Workflow Release Dashboard
+
+**Project Description:**
+
+The aim of this project is to create a dashboard for release testers and team leads on the Oppia developer team. The dashboard should be built as a standard App Engine app that supports login. It should show a list of all commits in develop. For each commit, it should show the status of the release candidate (RC) that was built for that commit:
+* A link to the RC (note: you can find links at the bottom of runs, e.g. https://github.com/oppia/release-scripts/actions/runs/3987166674).
+* Did that RC pass acceptance testing, fail acceptance testing, or get skipped?
+* For all RCs that pass acceptance testing:
+  - **Prod server**. Show a button next to each deployable RC to deploy it to prod. An RC is deployable if it is strictly newer (here “newer” refers to commit order, not the timestamp on the commit) than the currently-deployed RC. Deploying should be restricted to members of the team-leads-and-release-coordinators Google Group or allowlist; when clicked, it should notify that mailing list. The deployed version should be formatted on App Engine as {{tag}}-{{commithash}}, where {{tag}} is the release version (e.g. “3.2.1”).
+  - **Test server**.
+    - If the RC is currently provisioned on a test server, show a link to that test server.
+    - If not, show a button to provision a test server on BrowserStack with this RC. This button should be restricted to members of the release-testers Google Group or allowlist. Once provisioned, it should send an email to the requester and the status in the dashboard should be updated. The deployed version should be formatted as {{tag}}-{{commithash}}, where {{tag}} is the release version (e.g. “3.2.1”).
+
+(Note that the team-leads-and-release-coordinators Google Group will be a member of the release-testers Google Group.)
+
+Pagination should be included to allow paging through multiple pages of commits (50 per page), but it is fine not to go back more than 300 commits or so.
+
+Include full documentation on how to update code, test, and deploy new releases of this app. The app will also need unit tests and linting, and to support easy developer setup.
+
+**Size of this project:** large (~350 hours)
+
+**Potential Mentors:** 
+
+**Knowledge/Skills Recommended:**
```

Comment:
I just added these to the idea doc:

* Frontend development (probably Angular, but we don’t require that)
* Backend development (probably Python, but we don’t require that)